### ISSUE TITLE
on GME-Creation: could not find any of these files...

### ISSUE THREAD

**Issue by Dani22m** [thawn/ttmp32gme#35]
Hi,
if i click on "create GEM" i got the following error for every file in the album and no GME-File will be created.

```
'C:\Temp\ttmp32gme\library\1\3.ogg': Invalid argument
ffmpeg version 3.2.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.4.0 (GCC)
  configuration: --enable-gpl --enable-version3 --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
  libavutil      55. 34.100 / 55. 34.100
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.100 / 57. 56.100
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
'C:\Temp\ttmp32gme\library\1\4.ogg': Invalid argument
C:\Users\Daniel\AppData\Local\Temp\par-44616e69656c\cache-d4cd2848b70ff6e5ca5e050927fb42b31959a03d\inc\lib\tttool.exe --dpi 1200 --pixel-size 2 assemble 1.yaml
[ERROR] Could not find any of these files:
.\audio/track_0.wav
.\audio/track_0.ogg
.\audio/track_0.flac
.\audio/track_0.mp3
```

ttmp32gme configuration:
ttmp32gme library path: C:\Temp\ttmp32gme\library

where should be the folder "\audio" ?

**Comment by thawn** [comment]
What file format are you using?
MP3 or ogg?
also, in the configuration, did you by chance change the format from mp3 to ogg?

**Comment by Dani22m** [comment]
Hi, i try it with both File types. And i have change the File type in the configuration.

**Comment by thawn** [comment]
Could you please send me the files that cause the problem, so I can reproduce the issue?

**Comment by Dani22m** [comment]
Yes, I've attach the files
[samples.zip](https://github.com/thawn/ttmp32gme/files/4530900/samples.zip)

**Comment by thawn** [comment]
@Dani22m sorry for the long delay. I just tested on my system and the samples you sent worked fine for me (with both mp3 and ogg settings in the configuration). Did you change the "library path" in the configuration?

**Comment by rmoro-de** [comment]
the some issue
by convert to gme with lame OGG


Use of uninitialized value $ENV{"HOME"} in concatenation (.) or string at Music/Tag.pm line 1066, <DATA> line 1.
Updating config...
Update successful.
[MSG] Server running on port: 10020
Open http://127.0.0.1:10020/ in your favorite web browser to continue.

[MSG] using tttool: C:\Users\lenovo\AppData\Local\Temp\par-6c656e6f766f\cache-d4cd2848b70ff6e5ca5e050927fb42b31959a03d\inc\lib\tttool.exe
copying albums to library
[ERROR] WARNING: id3 tag missing or incomplete for C:\Users\lenovo\AppData\Roaming\ttmp32gme\library\temp\1\Vera F Birkenbihl - Denk-Strategien 1v4.mp3.
Please add an id3v2 tag containing at least album, title and track number to your mp3 file in order to get proper album and track info.
deleting C:\Users\lenovo\AppData\Roaming\ttmp32gme\library\temp
unhandled callback exception on event (/assets/images/920/cover.jpeg, AnyEvent::HTTPD=HASH(0x9223060), AnyEvent::HTTPD::Request=HASH(0x9413f18)): Can't read C:\Users\lenovo\AppData\Roaming\ttmp32gme\library\Bajki_Grajki\cover.jpeg: No such file or directory at TTMp32Gme/LibraryHandler.pm line 172.

ffmpeg version 3.2.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.4.0 (GCC)
  configuration: --enable-gpl --enable-version3 --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
  libavutil      55. 34.100 / 55. 34.100
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.100 / 57. 56.100
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
'C:\Users\lenovo\AppData\Roaming\ttmp32gme\library\2\1.mp3': Invalid argument
ffmpeg version 3.2.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.4.0 (GCC)
  configuration: --enable-gpl --enable-version3 --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
  libavutil      55. 34.100 / 55. 34.100
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.100 / 57. 56.100
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
'C:\Users\lenovo\AppData\Roaming\ttmp32gme\library\2\Vera_F_Birkenbihl_-_Denk-Strategien_1v4.mp3': Invalid argument
ffmpeg version 3.2.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.4.0 (GCC)
  configuration: --enable-gpl --enable-version3 --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
  libavutil      55. 34.100 / 55. 34.100
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.100 / 57. 56.100
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
'C:\Users\lenovo\AppData\Roaming\ttmp32gme\library\2\2.mp3': Invalid argument
C:\Users\lenovo\AppData\Local\Temp\par-6c656e6f766f\cache-d4cd2848b70ff6e5ca5e050927fb42b31959a03d\inc\lib\tttool.exe --dpi 1200 --pixel-size 2 assemble 2.yaml
[ERROR] Could not find any of these files:
.\audio/track_0.wav
.\audio/track_0.ogg
.\audio/track_0.flac
.\audio/track_0.mp3

Could not find any of these files:
.\audio/track_1.wav
.\audio/track_1.ogg
.\audio/track_1.flac
.\audio/track_1.mp3

Could not find any of these files:
.\audio/track_2.wav
.\audio/track_2.ogg
.\audio/track_2.flac
.\audio/track_2.mp3



works fine with MP3

**Comment by thawn** [comment]
Thank you very much for posting the error messages. It looks like there is an issue with converting mp3s that don't have a proper id3 tag to ogg. Particularly album, title and track number are important.

In principle, ttmp32gme should also work without id3 tag, but it seems that is not the case. I will fix this bug in the next release, if I manage to reproduce it on my test systems (which I have not managed so far).

Until that happens, please add album, title and track number to the id3 tag of the mp3s before uploading them - with that it should work.

**Comment by derpicknicker1** [comment]
In my case here are two issues with the conversion to `.ogg` files:
1.  On windows the file-name-arguments in the conversion command have to be in double quotation marks instead of single quotation marks. The single quotation marks give me the same issues as described above (file not found). https://github.com/thawn/ttmp32gme/blob/8a3c369312dc4637b1519cf32f6c0df49a4c89e1/src/TTMp32Gme/TttoolHandler.pm#L104
should be ==> `` `$ff_command -i "$source_file" -ar 22050 -ac 1 "$target_file"`; ``
But honestly I don't know if this breaks behaviour on linux/mac.

2.  If there is a cover art integrated in the `.mp3` file, ffmpeg tries to convert not only audio stream but also a video stream. This "breaks" the resulting `.ogg` file in a way, that the tiptoi can not play the `.gme`
**Solution:** use `-map a` as parameter for ffmpeg to tell it to extract only audio data. (  `` `$ff_command -map a -i "$source_file" -ar 22050 -ac 1 "$target_file"`; `` )

**Comment by thawn** [comment]
fixed in release v1.0.0
